File: backend/amoz/src/main/java/com/zpi/amoz/requests/UserRegisterRequest.java
package com.zpi.amoz.requests;

import com.zpi.amoz.enums.Sex;
import com.zpi.amoz.models.Person;
import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.Valid;
import jakarta.validation.constraints.*;

import java.time.LocalDate;
import java.util.Optional;

@Schema(description = "Request do rejestracji użytkownika, zawierający dane osobowe oraz dane kontaktowe osoby kontaktowej.")
public record UserRegisterRequest(

        @Schema(description = "Dane osobowe użytkownika", implementation = PersonCreateRequest.class)
        @NotNull(message = "Personal information is required")
        @Valid
        PersonCreateRequest person,

        @Schema(description = "Dane kontaktowe osoby kontaktowej", implementation = ContactPersonCreateRequest.class)
        @NotNull(message = "Contact person details are required")
        @Valid
        ContactPersonCreateRequest contactPerson
) {
}

File: frontend/app/src/main/java/com/example/amoz/api/requests/UserRegisterRequest.kt
package com.example.amoz.api.requests

import com.example.amoz.interfaces.ValidatableRequest
import com.example.amoz.models.ContactPerson
import com.example.amoz.models.Employee
import com.example.amoz.models.Person
import com.example.validation.annotations.NotNullable
import kotlinx.serialization.Serializable

@Serializable
data class UserRegisterRequest(
    @field:NotNullable(nameOfField = "Person")
    val person: PersonCreateRequest,

    @field:NotNullable(nameOfField = "Contact person")
    val contactPerson: ContactPersonCreateRequest
) : ValidatableRequest<UserRegisterRequest>() {
    constructor(person: Person, contactPerson: ContactPerson): this(
        person = PersonCreateRequest(person),
        contactPerson = ContactPersonCreateRequest(contactPerson)
    )

    constructor(employee: Employee): this(
        person = employee.person,
        contactPerson = employee.contactPerson
    )
}
